hal/armv7m/imxrt/10xx/hal.c

```diff
@@ -43,18 +43,13 @@ extern void console_init(void);
 void hal_init(void)
 {
 	_imxrt_init();
+	_imxrt_scbSetPriorityGrouping(3);
+
 	mpu_init();
 	timer_init();
-
 	console_init();
 
-	syspage_init();
-	syspage_setAddress(ADDR_SYSPAGE);
-
-	_imxrt_scbSetPriorityGrouping(3);
-
-	/* Add entries related to plo image */
-	syspage_addEntries((addr_t)_plo_bss, (addr_t)_end - (addr_t)_plo_bss + SIZE_STACK);
+	hal_common.halSyspage = (syspage_hal_t *)ADDRESS_SYSPAGE;
```

Comment:
Why can't hal get a pointer from outside?

Reply:
Because the location of SYSPAG is hardware dependent and this is responsibility of HAL.

Reply:
So maybe HAL should set address of the whole syspage, not only of syspage_hal_t part?

Reply:
I agree, but let me evaluate all changes.